tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/HotChocolateTests.cs

```diff
@@ -0,0 +1,261 @@
+// <copyright file="HotChocolateTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#if NET5_0_OR_GREATER || NETCOREAPP3_1_OR_GREATER
+
+#pragma warning disable SA1402 // File may only contain a single type
+#pragma warning disable SA1649 // File name should match first type name
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.TestHelpers;
+using VerifyXunit;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Datadog.Trace.ClrProfiler.IntegrationTests
+{
+    public class HotChocolate12Tests : HotChocolateTests
+    {
+        public HotChocolate12Tests(ITestOutputHelper output)
+            : base("HotChocolate", output, nameof(HotChocolate12Tests))
+        {
+        }
+
+        [SkippableTheory]
+        [MemberData(nameof(PackageVersions.HotChocolate), MemberType = typeof(PackageVersions))]
+        [Trait("Category", "EndToEnd")]
+        [Trait("RunOnWindows", "True")]
+        [Trait("SupportsInstrumentationVerification", "True")]
+        public async Task SubmitsTraces(string packageVersion)
+            => await RunSubmitsTraces(packageVersion);
+    }
+
+    [UsesVerify]
+    public abstract class HotChocolateTests : TestHelper
+    {
+        private const string ServiceVersion = "1.0.0";
+
+        private readonly string _testName;
+
+        protected HotChocolateTests(string sampleAppName, ITestOutputHelper output, string testName)
+            : base(sampleAppName, output)
+        {
+            SetServiceVersion(ServiceVersion);
+
+            _testName = testName;
+        }
+
+        protected async Task RunSubmitsTraces(string packageVersion = "")
+        {
+            SetInstrumentationVerification();
+            using var telemetry = this.ConfigureTelemetry();
```

Comment:
Note to self: need to review the usages of `ConfigureTelemetry()` - it shouldn't be necessary, `EnvironmentHelper.GetMockAgent()` should configure telemetry automatically